modules/transfers/src/infrastructure/redis_transfer_repo.ts

```diff
@@ -152,7 +155,23 @@ export class RedisTransferStateRepo implements ITransfersRepo {
           this._logger.error('Unsuccessful attempt to store the entity state in redis - for key: ' + key)
           return reject(err)
         }
-        return resolve()
+
+        if (this._expirationInSeconds > 0) {
```

Comment:
we can use this._redisClient.setex(key, ttl, value, ..), this way it will be atomic

Reply:
Awesome, I looked through the documentation and was not able to find this anywhere. Thanks.